src/security/ZeroSensitiveMemory.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 1996-2021 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_SECURITY_ZEROSENSITIVEMEMORY_H
+#define SQUID_SRC_SECURITY_ZEROSENSITIVEMEMORY_H
+
+#if HAVE_LIMITS_H
+#include <limits.h>
+#endif
+
+#if HAVE_STRING_H
+#include <string.h>
+#endif
+
+#include <stdexcept>
+
+namespace Security {
+
+inline static void
+ZeroSensitiveMemory(void *dst, const size_t len)
+{
+    if (!dst)
+        throw std::runtime_error("Cannot clear a null buffer");
+    if (len > SIZE_MAX)
+        throw std::runtime_error("Cannot clear a buffer of length exceeding SIZE_MAX");
```

Comment:
ok then I guess code user will be responsible enough ;-)

Reply:
Well, the caller may still make mistakes like feeding `-1` or a huge value to us. My point was that the function itself _cannot_ (as in "does not have the ability to") detect those mistakes (because the compiler will convert them into valid (from the function point of view) values). In many cases, such conversions will happen with a compiler warning that the developer will notice and address, but that is a different story.

Thank you for removing this non-working check. This change request is addressed.